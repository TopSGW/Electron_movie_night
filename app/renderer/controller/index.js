// Logic for mapping application state onto component props.
import {
  App,
  DisplayMoviesContainer,
  ImportMovies,
  ImportStats,
  SearchMovies,
  MainContent,
  MovieThumbnailContainer
} from './containers'

// Logic for mapping application events directly to redux dispatch actions.
import './mapEventsToDispatch'

// Re-export all the containers.
export {
  App,
  DisplayMoviesContainer,
  ImportMovies,
  ImportStats,
  MainContent,
  MovieThumbnailContainer,
  SearchMovies
}
