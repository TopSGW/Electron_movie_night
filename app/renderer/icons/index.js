import cn from 'classnames'
import React from 'react'

const Icon = ({ className, border, fixedWidth, large, pulse, size3x, ...props }) => {
  return <i
    className={cn(
      'fa',
      className,
      pulse && 'fa-pulse',
      large && 'fa-lg',
      fixedWidth && 'fa-fw',
      size3x && 'fa-3x',
      border && 'fa-border'
    )}
    {...props}
  />
}

export const Anchor = ({ className, ...props }) => {
  return <Icon className={cn('fa-anchor', className)} {...props} />
}

export const Angle = ({ className, down, left, right, up, ...props }) => {
  return <Icon
    className={cn(
      down && 'fa-angle-down',
      left && 'fa-angle-left',
      right && 'fa-angle-right',
      up && 'fa-angle-up',
      !down && !left && !right && !up && 'fa-angle-right', // default
      className
    )}
    {...props}
  />
}

export const Bars = ({ className, ...props }) => {
  return <Icon className={cn('fa-bars', className)} {...props} />
}

export const Ban = ({ className, ...props }) => {
  return <Icon className={cn('fa-ban', className)} {...props} />
}

export const Chevron = ({ className, down, left, right, up, ...props }) => {
  return <Icon
    className={cn(
      down && 'fa-chevron-down',
      left && 'fa-chevron-left',
      right && 'fa-chevron-right',
      up && 'fa-chevron-up',
      !down && !left && !right && !up && 'fa-chevron-right', // default
      className
    )}
    {...props}
  />
}

export const Database = ({ className, ...props }) => {
  return <Icon className={cn('fa-database', className)} {...props} />
}

export const Close = ({ className, ...props }) => {
  return <Icon className={cn('fa-close', className)} {...props} />
}

export const Edit = ({ className, ...props }) => {
  return <Icon className={cn('fa-pencil', className)} {...props} />
}

export const FileIcon = ({ className, video, ...props }) => {
  return <Icon
    className={cn(
      video && 'fa-file-video-o',
      !video && 'fa-file', // default
      className
    )}
    {...props}
  />
}

export const ImportIcon = ({ className, ...props }) => {
  return <Icon
    className={cn(
      'fa-download',
      className
    )}
    {...props}
  />
}

export const Info = ({ className, circle, ...props }) => {
  return <Icon
    className={cn(
      circle && 'fa-info-circle',
      !circle && 'fa-info',
      className
    )}
    {...props}
  />
}

export const Play = ({ className, ...props }) => {
  return <Icon className={cn('fa-play', className)} {...props} />
}

export const Search = ({ className, ...props }) => {
  return <Icon className={cn('fa-search', className)} {...props} />
}

export const Spinner = ({ className, ...props }) => {
  return <Icon className={cn('fa-spinner', className)} {...props} />
}

export const Star = ({ className, half, outline, ...props }) => {
  return <Icon
    className={cn(
      half && 'fa-star-half-o',
      outline && 'fa-star-o',
      !half && !outline && 'fa-star', // default
      className
    )}
    {...props}
  />
}

export const Trash = ({ className, ...props }) => {
  return <Icon className={cn('fa-trash', className)} {...props} />
}
