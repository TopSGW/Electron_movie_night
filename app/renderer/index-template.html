<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <title>Movie Night</title>
    <link rel="stylesheet" href="../vendor/font-awesome-4.7.0/css/font-awesome.min.css">
    <%=htmlWebpackPlugin.options.css%>
  </head>
  <body style="background-color:rgb(20,20,20) !important; /* override bootstrap */">
    <div id="root"></div>
    <script src="<%=htmlWebpackPlugin.options.bundle%>"></script>
  </body>
</html>
