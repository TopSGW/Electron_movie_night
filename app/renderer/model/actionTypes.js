'use strict'

// Crawl actions
export const CRAWL_DIRECTORY = 'crawl-directory'
export const IS_CRAWLING = 'is-crawling'
export const SHOW_CRAWL_STATS = 'show-crawl-state'
export const SET_CRAWL_STATS = 'set-crawl-stats'

// Database loaded actions
export const DATABASE_LOADED = 'database-loaded'

// Featured movie actions
export const CLEAR_FEATURED_MOVIE = 'clear-featured-movie'
export const UPDATE_FEATURED_MOVIE = 'update-featured-movie'

// Movie database actions
export const UPDATE_MOVIE_DATABASE = 'update-movie-database'

// Search actions
export const CLEAR_SEARCH_RESULTS = 'clear-search-results'
export const CLEAR_SEARCH_QUERY = 'clear-search-query'
export const UPDATE_SEARCH_CATEGORY = 'update-search-category'
export const UPDATE_SEARCH_QUERY = 'update-search-query'
