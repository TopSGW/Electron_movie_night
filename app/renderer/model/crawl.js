'use strict'
import {
  CRAWL_DIRECTORY,
  IS_CRAWLING,
  SHOW_CRAWL_STATS,
  SET_CRAWL_STATS
} from './actionTypes'

const initialState = {
  directory: '',
  active: false,
  showCrawlStatsOverlay: false,
  moviesFound: 0,
  inProgress: []
}

export default (state = initialState, action) => {
  switch (action.type) {
    case CRAWL_DIRECTORY: {
      return {
        ...state,
        directory: action.payload
      }
    }

    case IS_CRAWLING: {
      return {
        ...state,
        active: action.payload
      }
    }

    case SHOW_CRAWL_STATS: {
      return {
        ...state,
        showCrawlStatsOverlay: !!action.payload
      }
    }

    case SET_CRAWL_STATS: {
      const { moviesFound, inProgress } = action.payload
      return {
        ...state,
        moviesFound,
        inProgress
      }
    }

    default:
      return state
  }
}

export const active = (state) => state.active
export const directory = (state) => state.directory
export const showCrawlStatsOverlay = (state) => state.showCrawlStatsOverlay
export const crawlStats = (state) => ({
  active: state.active,
  crawlDirectory: state.directory,
  moviesFound: state.moviesFound,
  inProgress: state.inProgress
})
