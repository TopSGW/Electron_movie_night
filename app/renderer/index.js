import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'

import { logEnv } from '../shared/utils'
import { App } from './controller'
import store from './model'
import logger from './mainWindowLogger'

logEnv(logger)

// Render the App.
ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root')
)
