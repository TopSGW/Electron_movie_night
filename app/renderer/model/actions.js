'use strict'
import * as ACTIONS from './actionTypes'

// Database loaded actions
// -----------------------
export function databaseLoaded () {
  return {
    type: ACTIONS.DATABASE_LOADED
  }
}

// Directory crawl actions
// -----------------------
export function updateCrawlState (isCrawling) {
  return {
    type: ACTIONS.IS_CRAWLING,
    payload: isCrawling
  }
}

export function updateCurrentCrawlDirectory (directory) {
  return {
    type: ACTIONS.CRAWL_DIRECTORY,
    payload: directory
  }
}

export function updateCrawlStats ({ moviesFound, inProgress }) {
  return {
    type: ACTIONS.SET_CRAWL_STATS,
    payload: { moviesFound, inProgress }
  }
}

export function showCrawlStats () {
  return {
    type: ACTIONS.SHOW_CRAWL_STATS,
    payload: true
  }
}

export function hideCrawlStats () {
  return {
    type: ACTIONS.SHOW_CRAWL_STATS,
    payload: false
  }
}

// Movie database actions
// -----------------------
export function updateMovieDB (movieDB) {
  return {
    type: ACTIONS.UPDATE_MOVIE_DATABASE,
    payload: movieDB
  }
}

// Featured Movie actions
// -----------------------
export function clearFeaturedMovie () {
  return {
    type: ACTIONS.CLEAR_FEATURED_MOVIE
  }
}

export function updateFeaturedMovie ({ movie, action, panelID }) {
  return {
    type: ACTIONS.UPDATE_FEATURED_MOVIE,
    payload: { movie, action, panelID }
  }
}

// Search actions
// ---------------
export function clearSearchResults () {
  return {
    type: ACTIONS.CLEAR_SEARCH_RESULTS
  }
}

export function clearSearchQuery () {
  return {
    type: ACTIONS.CLEAR_SEARCH_QUERY
  }
}

export function updateSearchCategory (text) {
  return {
    type: ACTIONS.UPDATE_SEARCH_CATEGORY,
    payload: text
  }
}

export function updateSearchQuery (text) {
  return {
    type: ACTIONS.UPDATE_SEARCH_QUERY,
    payload: text
  }
}
