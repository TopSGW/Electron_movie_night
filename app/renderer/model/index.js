import { ipcRenderer } from 'electron'
import { createStore, applyMiddleware, combineReducers } from 'redux'
import reduxLogger from 'redux-logger'

import { MOVE_MOVIE_TO_TRASH, UPDATE_MOVIE_METADATA } from '../../shared/events'
import { isDevEnv } from '../../shared/utils'
import appLogger from '../mainWindowLogger'

// Reducers
import dbLoaded from './dbLoaded'
import crawl from './crawl'
import featuredMovie from './featuredMovie'
import movies from './movies'
import search from './search'

// Top level state selectors.
import {
  crawlActiveSelector,
  crawlStatsSelector,
  showCrawlStatsSelector,
  movieDBSelector,
  searchCategorySelector,
  searchQuerySelector
} from './selectors'

// Memoized Selectors
import enhancedFeaturedMovieSelector from './enhancedFeaturedMovie'
import visibleMoviesSelector from './visibleMovies'

// Re-export all the action creators.
// ----------------------------------
export * from './actions'

// Electron action creators.
// -------------------------
export const updateMetadataFor = (movie) => {
  ipcRenderer.send(UPDATE_MOVIE_METADATA, movie)
  appLogger.info('Sent UPDATE_MOVIE_METADATA event', { movie: movie.title })
}

export const deleteMovie = (movie) => {
  ipcRenderer.send(MOVE_MOVIE_TO_TRASH, movie)
  appLogger.info(`Sent ${MOVE_MOVIE_TO_TRASH} event`, { movie: movie.title })
}

// Public selectors.
// These allow connected containers to depend on state
// without requiring knowledge of the state's shape.
// ----------------------------------------------------------------------
export const getAllMovies = (state) => {
  return movieDBSelector(state)
}

export const getCrawlActive = (state) => {
  return crawlActiveSelector(state)
}

export const getCrawlStats = (state) => {
  return crawlStatsSelector(state)
}

export const getShowCrawlStats = (state) => {
  return showCrawlStatsSelector(state)
}

export const getFeaturedMovie = (state) => {
  return enhancedFeaturedMovieSelector(state)
}

export const getSearchCategory = (state) => {
  return searchCategorySelector(state)
}

export const getSearchQuery = (state) => {
  return searchQuerySelector(state)
}

export const getVisibleMovies = (state) => {
  return visibleMoviesSelector(state)
}

// Export function for creating redux store.
// -----------------------------------------
export const createReduxStore = () => {
  // Create the redux store.
  // This is the default export.
  // ---------------------------
  let middlewares = []
  if (isDevEnv()) {
    middlewares.push(reduxLogger)
  }

  const rootReducer = combineReducers({
    crawl,
    dbLoaded,
    featuredMovie,
    movies,
    search
  })

  const store = createStore(rootReducer, applyMiddleware(...middlewares))
  return store
}

// This is the actual store used in the app.
// -----------------------------------------
const store = createReduxStore()
export default store
