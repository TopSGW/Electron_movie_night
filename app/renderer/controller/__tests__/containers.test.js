'use strict'
/* globals jest, describe, test, expect */

import React from 'react'
import renderer from 'react-test-renderer'

import { createReduxStore } from '../../model'
import {
  App, mapStateToAppProps,
  ImportMovies, mapStateToImportMoviesProps,
  mapDispatchToMovieThumbnailProps,
  SearchMovies, mapStateToSearchBarProps, mapDispatchToSearchBarProps,
  DisplayMoviesContainer, mapStateToDisplayMoviesProps, mapDispatchToDisplayMoviesProps,
  MainContent, mapStateToMainContentProps
} from '../containers'

const store = createReduxStore()

const mockDispatch = jest.fn()
const lastDispatchResult = () => {
  const lastCall = mockDispatch.mock.calls.length - 1
  return mockDispatch.mock.calls[lastCall][0]
}

describe('App', () => {
  test('renders', () => {
    renderer.create(
      <App store={store} />
    )
  })

  test('mapStateToAppProps', () => {
    const result = mapStateToAppProps(store.getState())
    expect(result).toMatchSnapshot()
  })
})

describe('ImportMovies', () => {
  test('renders', () => {
    renderer.create(
      <ImportMovies store={store} />
    )
  })

  test('mapStateToImportMoviesProps', () => {
    const result = mapStateToImportMoviesProps(store.getState())
    expect(result).toMatchSnapshot()
  })
})

describe('SearchMovies', () => {
  test('renders', () => {
    renderer.create(
      <SearchMovies store={store} />
    )
  })

  test('mapStateToSearchBarProps', () => {
    const result = mapStateToSearchBarProps(store.getState())
    expect(result).toMatchSnapshot()
  })

  test('mapDispatchToSearchBarProps', () => {
    const result = mapDispatchToSearchBarProps(mockDispatch)
    expect(result).toMatchSnapshot()

    result.handleQueryChange('top gun')
    expect(lastDispatchResult()).toMatchSnapshot()

    result.handleClear()
    expect(lastDispatchResult()).toMatchSnapshot()
  })
})

describe('MovieThumbnailContainer', () => {
  test('mapDispatchToMovieThumbnailProps', () => {
    const result = mapDispatchToMovieThumbnailProps(mockDispatch)
    expect(result).toMatchSnapshot()

    result.handleMovieSelected({movie: 'top gun', action: 'click', panelID: 1})
    expect(lastDispatchResult()).toMatchSnapshot()
  })
})

describe('DisplayMoviesContainer', () => {
  test('renders', () => {
    renderer.create(
      <DisplayMoviesContainer store={store} />
    )
  })

  test('mapStateToDisplayMoviesProps', () => {
    const result = mapStateToDisplayMoviesProps(store.getState())
    expect(result).toMatchSnapshot()
  })

  test('mapDispatchToDisplayMoviesProps', () => {
    const result = mapDispatchToDisplayMoviesProps(mockDispatch)
    expect(result).toMatchSnapshot()

    result.clearSearchQuery()
    expect(lastDispatchResult()).toMatchSnapshot()

    result.clearFeaturedMovie()
    expect(lastDispatchResult()).toMatchSnapshot()

    result.updateSearchCategory('action')
    expect(lastDispatchResult()).toMatchSnapshot()
  })
})

describe('MainContent', () => {
  test('renders', () => {
    renderer.create(
      <MainContent store={store} />
    )
  })

  test('mapStateToMainContentProps', () => {
    const result = mapStateToMainContentProps(store.getState())
    expect(result).toMatchSnapshot()
  })
})
