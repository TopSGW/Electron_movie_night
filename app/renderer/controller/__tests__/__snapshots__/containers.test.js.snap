// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App mapStateToAppProps 1`] = `
Object {
  "dbLoaded": false,
  "showCrawlStatsOverlay": false,
}
`;

exports[`DisplayMoviesContainer mapDispatchToDisplayMoviesProps 1`] = `
Object {
  "clearFeaturedMovie": [Function],
  "clearSearchQuery": [Function],
  "deleteMovieFromDb": [Function],
  "updateMovieMetadata": [Function],
  "updateSearchCategory": [Function],
}
`;

exports[`DisplayMoviesContainer mapDispatchToDisplayMoviesProps 2`] = `
Object {
  "type": "clear-search-query",
}
`;

exports[`DisplayMoviesContainer mapDispatchToDisplayMoviesProps 3`] = `
Object {
  "type": "clear-featured-movie",
}
`;

exports[`DisplayMoviesContainer mapDispatchToDisplayMoviesProps 4`] = `
Object {
  "payload": "action",
  "type": "update-search-category",
}
`;

exports[`DisplayMoviesContainer mapStateToDisplayMoviesProps 1`] = `
Object {
  "featuredMovie": Object {
    "action": "",
    "movie": null,
    "panelID": -1,
  },
  "movies": Array [],
  "searchCategory": "",
  "searchQuery": "",
}
`;

exports[`ImportMovies mapStateToImportMoviesProps 1`] = `
Object {
  "busy": false,
  "haveMovies": false,
}
`;

exports[`MainContent mapStateToMainContentProps 1`] = `
Object {
  "handleAddMediaClick": [Function],
  "isCrawling": false,
  "totalMovieCount": 0,
}
`;

exports[`MovieThumbnailContainer mapDispatchToMovieThumbnailProps 1`] = `
Object {
  "handleMovieSelected": [Function],
}
`;

exports[`MovieThumbnailContainer mapDispatchToMovieThumbnailProps 2`] = `
Object {
  "payload": Object {
    "action": "click",
    "movie": "top gun",
    "panelID": 1,
  },
  "type": "update-featured-movie",
}
`;

exports[`SearchMovies mapDispatchToSearchBarProps 1`] = `
Object {
  "handleClear": [Function],
  "handleQueryChange": [Function],
}
`;

exports[`SearchMovies mapDispatchToSearchBarProps 2`] = `
Object {
  "payload": "top gun",
  "type": "update-search-query",
}
`;

exports[`SearchMovies mapDispatchToSearchBarProps 3`] = `
Object {
  "type": "clear-search-results",
}
`;

exports[`SearchMovies mapStateToSearchBarProps 1`] = `
Object {
  "searchCategory": "",
  "searchQuery": "",
}
`;
